pkg/autoscaler/metrics/collector_test.go

```diff
@@ -245,13 +256,16 @@ func TestMetricCollectorNoScraper(t *testing.T) {
 	wait.PollImmediate(10*time.Millisecond, 100*time.Millisecond, func() (bool, error) {
 		_, _, errCon := coll.StableAndPanicConcurrency(metricKey, now)
 		_, _, errRPS := coll.StableAndPanicRPS(metricKey, now)
-		return errRPS == ErrNoData && errCon == errRPS, nil
+		return errRPS == nil && errCon == nil, nil
 	})
```

Comment:
This wait is now redundant I think? Won't the error always be nil?

Reply:
Yes, it was redundant. I removed the waith and I moved the "noData" case to its own test function in the new commits.